freqtrade/strategy/interface.py

```diff
@@ -742,6 +768,16 @@ def advise_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
         :return: a Dataframe with all mandatory indicators for the strategies
         """
         logger.debug(f"Populating indicators for pair {metadata.get('pair')}.")
+
+        # call populate_indicators_Nm() which were tagged with @informative decorator.
+        for attr_name in dir(self.__class__):
+            cls_attr = getattr(self.__class__, attr_name)
+            if not callable(cls_attr):
+                continue
+            if not getattr(cls_attr, '_is_informative', False):
+                continue
+            dataframe = cls_attr(self, dataframe, metadata)
```

Comment:
I'd avoid iterating the whole class as much as possible.

As mentioned above - when we gather the "required" pairs at startup - we can also store the list of "informative indicator methods" in a list - which we can iterate here quite easily without doing a lot of unnecessary looping during runtime (the final list of methods to call will not change during runtime, so gathering it once should be fine).

Reply:
I think you've missed this part.
While you do keep `_extra_informative_pairs` now - you can also keep the method itself in a list - so it's easy to be called without tripple-checking and iterating through the whole class.

Reply:
I am no sure there is any reason for concern. Yes - it iterates all attributes of the class. This is also done on startup and takes nanoseconds so why care? I do agree "_is_informative" tag is rather dirty, but there is no better way to do it. Decorator code runs before class is created, so there is no way to get access to class object and store tagged methods in it. Therefore we must tag all methods and do this post-processing step after class is created.